### PR TITLE
Update controllers to v33

### PR DESCRIPTION
**Pull request by Gudahtt**
The controllers package has been updated to v33. This release included various breaking changes relating to NFTs. All controller APIs and state have been updated to use the term "NFT" instead of "collectible".

Note that this PR is pointing at https://github.com/MetaMask/metamask-mobile/pull/5200, so I have skipped most of the steps in the PR template because I am unsure how or whether they apply.

### CODE REVIEW

File: app/components/UI/Transactions/index.test.tsx [hunk 1]
```diff
@@ -29,8 +29,8 @@ const initialState = {
       TokenRatesController: {
         contractExchangeRates: {},
       },
-      CollectiblesController: {
-        allCollectibleContracts: { '0x0': { 1: [] } },
+      NftController: {
+        allNfts: { '0x0': { 1: [] } },
```
Comment (tommasini) [inline app/components/UI/Transactions/index.test.tsx:33]: Should this be `allNftsContracts`?

**Review by tommasini — CHANGES_REQUESTED** [review]

File: app/components/UI/Transactions/index.test.tsx (hunk 1 above)
Comment (tommasini) [reply app/components/UI/Transactions/index.test.tsx:33]: I think this is the line that are doing the CI tests failing

File: app/components/UI/CollectibleContracts/index.js [hunk 2]
```diff
@@ -253,16 +253,14 @@ const CollectibleContracts = ({
...
-          <View style={styles.emptyView}>
-            <CollectibleDetectionModal
-              onDismiss={dismissNftInfo}
-              navigation={navigation}
-            />
-          </View>
-        )}
+      {isMainNet(chainId) && !nftDetectionDismissed && !useNftDetection && (
+        <View style={styles.emptyView}>
+          <CollectibleDetectionModal
```
Comment (sethkfman) [inline app/components/UI/CollectibleContracts/index.js:258]: Will there be plans to update `CollectibleDetectionModal`?

**Review by tommasini — APPROVED** [review]
LGTM, awesome, thanks!

**Comment by Gudahtt** [comment]
I considered going down the route of renaming all of the variables/components/etc. in mobile, but there are A LOT. It would have blown up the scope of this update enormously. I think we should do this as a follow-up refactor task instead.

**Pull request merged**